ChattoApp/ChattoApp/Source/Chat View Controllers/ScrollToBottomButtonChatViewController.swift

```diff
@@ -0,0 +1,44 @@
+/*
+ The MIT License (MIT)
+
+ Copyright (c) 2015-present Badoo Trading Limited.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in
+ all copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+*/
+
+import UIKit
+
+final class ScrollToBottomButtonChatViewController: DemoChatViewController {
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        let button = UIBarButtonItem(
+            title: "Scroll To Bottom",
+            style: .plain,
+            target: self,
+            action: #selector(handleTapOnScrollToBottomButton)
+        )
+        self.navigationItem.rightBarButtonItem = button
+    }
+
+    @objc
+    private func handleTapOnScrollToBottomButton() {
+        self.scrollToItem(withId: self.chatItemCompanionCollection[self.chatItemCompanionCollection.count - 1].uid, position: .bottom, animated: true)
```

Comment:
"- 1" is confusing and unclear. also, what about "Out of bounds" protection (we modify the index still)?